clash-lib/src/Clash/Core/TysPrim.hs

```diff
@@ -148,15 +154,89 @@ intPrimTc, integerPrimTc, charPrimTc, stringPrimTc, wordPrimTc,
   int64PrimTc, word64PrimTc, floatPrimTc, doublePrimTc, naturalPrimTc,
   byteArrayPrimTc :: TyCon
 intPrimTc     = liftedPrimTC intPrimTyConName
+#if MIN_VERSION_base(4,17,0)
+integerPrimTc =
+  let name = integerPrimTyConName
+      uniq = nameUniq name
+      isDcNm = mkUnsafeSystemName (showt 'IS)
+                                  (getKey integerISDataConKey)
+      isDc = MkData { dcName = isDcNm
+                    , dcUniq = nameUniq isDcNm
+                    , dcTag  = 1
```

Comment:
I'm going to be real annoying and request a different formatting here as the current one wastes a lot of horizontal space (which is okay, until it is not)..

```suggestion
  let 
    name = integerPrimTyConName
    uniq = nameUniq name
    isDcNm = mkUnsafeSystemName (showt 'IS) (getKey integerISDataConKey)
    isDc = MkData
      { dcName = isDcNm
      , dcUniq = nameUniq isDcNm
      , dcTag  = 1
```

Reply:
Also, mini-rant: I've never understood why people like the `MkFoo` pattern in Haskell-land. It's _alright_ when you're constructing stuff, but it makes no sense at all when deconstructing IMO.